### PR TITLE
GUVNOR-2851, GUVNOR-2870, GUVNOR-2835: Library perspective

### PR DESCRIPTION
**Pull request by paulovmr**
* The creation of new resources must be observed by the Library perspective.

### CODE REVIEW

**Comment by paulovmr** [comment]
Part of an ensemble:

https://github.com/droolsjbpm/kie-wb-common/pull/655
https://github.com/droolsjbpm/drools-wb/pull/393
https://github.com/droolsjbpm/jbpm-form-modeler/pull/86
https://github.com/droolsjbpm/jbpm-designer/pull/543
https://github.com/droolsjbpm/kie-wb-distributions/pull/453

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-client/src/main/java/org/drools/workbench/screens/dtablexls/client/handlers/NewDecisionTableXLSHandler.java [hunk 1]
```diff
@@ -141,6 +142,7 @@ public void execute() {
                                      busyIndicatorView.hideBusyIndicator();
                                      presenter.complete();
                                      notifySuccess();
+                                     newResourceSuccessEvent.fire( new NewResourceSuccessEvent( path ) );
```
Comment (manstis) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-client/src/main/java/org/drools/workbench/screens/dtablexls/client/handlers/NewDecisionTableXLSHandler.java]: I have to ask; why introduce a new ```NewResourceSuccessEvent``` whereas most other observers of new, updated and deleted resources etc use ```ResourceAddedEvent``` (raised from Uberfire's ```IOWatchService```)? The "Library" will not see new resources created externally and pushed into the Workbench repository (e.g ,```git clone...```, update, ```git push...```)

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-client/src/test/java/org/drools/workbench/screens/dtablexls/client/handlers/NewDecisionTableXLSHandlerTest.java [hunk 2]
```diff
@@ -85,13 +90,20 @@
 
     @Before
     public void setup() {
+        newResourceSuccessEventMock = spy( new EventSourceMock<NewResourceSuccessEvent>() {
```
Comment (manstis) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-client/src/test/java/org/drools/workbench/screens/dtablexls/client/handlers/NewDecisionTableXLSHandlerTest.java]: You might be better just ```@Mock EventSourceMock<NewResourceSuccessEvent> newResourceSuccessEventMock;``` (I only comment as I also used to use a concrete instances of ```EventSourceMock``` and override the ```fire``` method - mocking it seemed to provide the exact same function without the boiler plate).

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-client/src/main/java/org/drools/workbench/screens/dtablexls/client/handlers/NewDecisionTableXLSHandler.java (hunk 1 above)
Comment (paulovmr) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-client/src/main/java/org/drools/workbench/screens/dtablexls/client/handlers/NewDecisionTableXLSHandler.java]: This 'NewResourceSuccessEvent' idea is to only be fired only for the logged user modifications in the workbench, so we can make updates in the UI when the user creates a new asset (e.g. update the breadcrumb when navigating to the recently added asset).

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-client/src/test/java/org/drools/workbench/screens/dtablexls/client/handlers/NewDecisionTableXLSHandlerTest.java (hunk 2 above)
Comment (paulovmr) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-client/src/test/java/org/drools/workbench/screens/dtablexls/client/handlers/NewDecisionTableXLSHandlerTest.java]: Done (for the other tests as well) :)

**Pull request merged**